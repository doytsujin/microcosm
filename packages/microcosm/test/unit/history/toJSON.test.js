import Microcosm from 'microcosm'

describe('History::toJSON', function() {
  const action = n => n

  it('includes the size of the tree', function() {
    const repo = new Microcosm({ debug: true })

    repo.push(action)
    repo.push(action)

    let json = JSON.parse(JSON.stringify(repo.history))

    expect(json.size).toEqual(2)
  })

  it('includes an id reference to the head', function() {
    const repo = new Microcosm({ debug: true })

    repo.push(action)

    let json = JSON.parse(JSON.stringify(repo.history))

    expect(json.head).toEqual(repo.history.head.id)
  })

  it('includes an id reference to the root', function() {
    const repo = new Microcosm({ debug: true })

    repo.push(action)

    let json = JSON.parse(JSON.stringify(repo.history))

    expect(json.root).toEqual(repo.history.root.id)
  })

  it('serializes the tree of actions', function() {
    const repo = new Microcosm({ debug: true })

    repo.push('one', true)
    let two = repo.push('two', true)
    repo.push('three', true)
    repo.history.checkout(two)
    repo.push('four', true)

    expect(repo.history).toMatchSnapshot()
  })
})
