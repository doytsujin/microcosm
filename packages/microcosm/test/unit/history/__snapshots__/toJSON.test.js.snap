// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History::toJSON serializes the tree of actions 1`] = `
Object {
  "head": undefined,
  "list": Array [
    Object {
      "payload": true,
      "status": "complete",
      "tag": "one",
    },
    Object {
      "payload": true,
      "status": "complete",
      "tag": "two",
    },
    Object {
      "payload": true,
      "status": "complete",
      "tag": "four",
    },
  ],
  "root": undefined,
  "size": 3,
  "tree": Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "payload": true,
            "status": "complete",
            "tag": "three",
          },
          Object {
            "children": Array [],
            "payload": true,
            "status": "complete",
            "tag": "four",
          },
        ],
        "payload": true,
        "status": "complete",
        "tag": "two",
      },
    ],
    "payload": true,
    "status": "complete",
    "tag": "one",
  },
}
`;
